src/Microsoft.OData.Core/WriterValidator.cs

```diff
@@ -35,9 +54,17 @@ internal WriterValidator(ODataMessageWriterSettings settings)
         /// <returns>The created instance.</returns>
         public IDuplicatePropertyNameChecker CreateDuplicatePropertyNameChecker()
         {
-            return settings.ThrowOnDuplicatePropertyNames
-                   ? (IDuplicatePropertyNameChecker)new DuplicatePropertyNameChecker()
-                   : (IDuplicatePropertyNameChecker)new NullDuplicatePropertyNameChecker();
+            return this.duplicatePropertyNameCheckerObjectPool.GetObject();
+        }
+
+        /// <summary>
+        /// Release a DuplicatePropertyNameChecker instance.
+        /// </summary>
+        /// <returns>The created instance.</returns>
+        public void ReleaseDuplicatePropertyNameChecker(IDuplicatePropertyNameChecker duplicatePropertyNameChecker)
+        {
+            duplicatePropertyNameChecker.Reset();

```

Comment:
Since we're releasing the duplicate name checker, we are not sure it's going to be used again. Maybe it might be more economical to reset it when it's being requested from the pool. Here I'm also making the assumption that resetting a newly created duplicate name checker has negligible cost.

Reply:
Typically, we should call 'Reset' in the GetDuplicatePropertyNameChecker()'

Reply:
Fixed